### PR TITLE
Consistent state handling of domain whois data.

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

When the contact information for a domain is changed on the Edit Contact Info page, the correct email that the verification has been sent to isn't reflected on the Domain Settings page because the two pages are using different data to display the email address.

#### Testing instructions

- Pick a domain that we manage the emails for.
- Update the email address in the Edit Contact Info page.
- Return to the Domain Settings page.
- Make sure that the ICANN notice at the top of the page shows the newly saved email.
- Make sure that the email is sent to the correct address.
- Make sure that the contact form still works correctly for newly purchased domains as well.

### CODE REVIEW

**Comment by hambai** [comment]
Well I tried to test this but I think it's still not working as it should. Here's what I did - I have a domain with a verified contact. I changed the email to something that is not verified and I see two issues. When I update the contact I get one of those notices:

Screenshot 2019-07-21 at 10 20 30

So now I'm still on the contact edit screen - I can click back twice and I'm on the domain settings screen but nothing shows up. I have to go to another page in order to get the notice that the email is not verified.

Then there is one more issue. When I update my contact email to something else again it correctly shows the new contact email in the domain settings notification, but when I click change email address I still have the old email in the contact form...

**Comment by hambai** [comment]
Oh and one more thing I just noticed. I updated my contact details to an email that is already verified. But I'm still shown the "Important: Verify Your Email Address" notification on the domain settings screen.

**Comment by delputnam** [comment]
Ok. I think I've got this one read for another look.  Thanks for the help, @hambai !

File: client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx [hunk 1]
```diff
@@ -134,17 +163,17 @@ class EditContactInfoFormCard extends React.Component {
...
-		const currentEmail = this.props.contactInformation.email,
+		const { email } = this.getContactFormFieldValues(),
 			wpcomEmail = this.props.currentUser.email,
 			strong = <strong />;
 
 		return (
 			<p>
 				{ this.props.translate(
-					'If you don’t have access to {{strong}}%(currentEmail)s{{/strong}}, ' +
+					'If you don’t have access to {{strong}}%(email)s{{/strong}}, ' +
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'If you don’t have access to {{strong}}%(currentEmail)s{{/strong}}, we will also email you at {{strong}}%(wpcomEmail)s{{/strong}}, as backup.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx [hunk 2]
```diff
@@ -178,15 +207,15 @@ class EditContactInfoFormCard extends React.Component {
...
 				{
-					args: { oldEmail: currentEmail, newEmail: newContactDetails.email },
+					args: { oldEmail: email, newEmail: newContactDetails.email },
 					components: { strong },
 				}
 			);
 		} else {
 			text = translate(
-				'We’ll email you at {{strong}}%(currentEmail)s{{/strong}} with a link to confirm the new details. ' +
+				'We’ll email you at {{strong}}%(email)s{{/strong}} with a link to confirm the new details. ' +
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'We’ll email you at {{strong}}%(currentEmail)s{{/strong}} with a link to confirm the new details. The change won't go live until we receive confirmation from this email.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by eltongo** [comment]
Tested and looks good to me.

**Comment by hambai** [comment]
So when I open a fresh contact form it sometimes comes empty.

It seems there is one last `fetchWhois` call in client/components/data/domain-management/index.jsx - I think we should clean it along all the related reducers, actions and etc. 

And also we should verify that the `EditContactInfo:isDataLoading` method is checking the correct whois state so that we don't render the form while the whois data is still fetching.

**Comment by eltongo** [comment]
So I have a domain with Privacy Protection disabled. So I went ahead and entered a new email address in my contact info, and after saving I clicked back once. Now I'm in the view that shows the Public Record Preview.

Here, my previous email address is listed instead of the new one I entered. Maybe it would make sense to display the previous email address if the new one is not verified yet (since it's a Public Record Preview), but the email address I entered was already verified.

Other than that I noticed no issues.

**Comment by delputnam** [comment]
> Here, my previous email address is listed instead of the new one I entered.

Thanks for catching this @eltongo!

I've updated the ContactDisplay component to use `requestWhois` to fetch the whois data as well.

**Comment by hambai** [comment]
Okay, I think I found one/few more issue/s - when you click on the `Contacts` button you go to this page:

Screenshot 2019-08-20 at 14 51 37

Notice the missing public record preview box.

Now when I click on edit, and I edit the contact and then go back - the public record box is displayed but it shows the wrong contact info (since privacy for this domain is enabled).

Screenshot 2019-08-20 at 14 52 01

**Pull request merged**